packages/config/src/__tests__/defineConfig.test.ts

```diff
@@ -0,0 +1,54 @@
+import { defineConfig } from '../defineConfig'
+import type { PrismaConfig, PrismaConfigEnvLoad } from '../prismaConfig'
+
+describe('defineConfig', () => {
+  describe('loadEnv', () => {
+    function assertKindLoad(
+      configEnv: PrismaConfig['env'],
+    ): asserts configEnv is PrismaConfigEnvLoad {
+      expect(configEnv.kind).toBe('load')
+    }
+
+    test('if no `loadEnv` function is provided, it should skip loading any environment variables', () => {
+      const config = defineConfig({})
+      expect(config.env).toEqual({ kind: 'skip' })
+    })
+
+    test('if a `loadEnv` function is provided, it should load environment variables using the provided function', async () => {
+      const config = defineConfig({
+        loadEnv: async () => {
+          return {
+            TEST_CONNECTION_STRING: 'postgresql://something',
+          }
+        },
+      })
+      expect(config.env).toEqual({
+        kind: 'load',
+        loadEnv: expect.any(Function),
+      })
+      assertKindLoad(config.env)
+      expect(await config.env.loadEnv()).toEqual({
+        TEST_CONNECTION_STRING: 'postgresql://something',
+      })
+    })
+  })
```

Comment:
In [ORM-607](https://linear.app/prisma-company/issue/ORM-607/when-prismaconfigts-is-detected-disable-automatic-env-file-loading), we should create a new fixture-based test in `@prisma/internals` that asserts how `process.env` changes when `prisma.config.ts` is defined.